test/robust.jl

```diff
@@ -0,0 +1,49 @@
+using StatsBase
+using Base.Test
+
+### Trimming outliers
+
+@test trim([1,2,3,4,5,6,7,8],   prop=0.1) == [1,2,3,4,5,6,7,8]
+@test trim([1,2,3,4,5,6,7,8],   prop=0.2) == [2,3,4,5,6,7]
+@test trim([1,2,3,4,5,6,7,8,9], prop=0.4) == [4,5,6]
+@test trim([1,2,3,4,5,6,7,8],   count=1)  == [2,3,4,5,6,7]
+@test trim([1,2,3,4,5,6,7,8,9], count=3)  == [4,5,6]
+
+
+@test_throws ArgumentError trim([])
+@test_throws ArgumentError trim([1,2,3,4,5], prop=0.5)
+
+@test trim!([1,2,3,4,5,6,7,8],   prop=0.1) == [1,2,3,4,5,6,7,8]
+@test trim!([1,2,3,4,5,6,7,8],   prop=0.2) == [2,3,4,5,6,7]
+@test trim!([1,2,3,4,5,6,7,8,9], prop=0.4) == [4,5,6]
+@test trim!([1,2,3,4,5,6,7,8],   count=1)  == [2,3,4,5,6,7]
+@test trim!([1,2,3,4,5,6,7,8,9], count=3)  == [4,5,6]
+
+@test_throws ArgumentError trim!([])
+@test_throws ArgumentError trim!([1,2,3,4,5], prop=0.5)
+
+@test winsor([1,2,3,4,5,6,7,8],   prop=0.1) == [1,2,3,4,5,6,7,8]
+@test winsor([1,2,3,4,5,6,7,8],   prop=0.2) == [2,2,3,4,5,6,7,7]
+@test winsor([1,2,3,4,5,6,7,8,9], prop=0.4) == [4,4,4,4,5,6,6,6,6]
+@test winsor([1,2,3,4,5,6,7,8],   count=1)  == [2,2,3,4,5,6,7,7]
+@test winsor([1,2,3,4,5,6,7,8,9], count=3)  == [4,4,4,4,5,6,6,6,6]
+
+@test_throws ArgumentError winsor([])
+@test_throws ArgumentError winsor([1,2,3,4,5], prop=0.5)
+
+@test winsor!([1,2,3,4,5,6,7,8],   prop=0.1) == [1,2,3,4,5,6,7,8]
+@test winsor!([1,2,3,4,5,6,7,8],   prop=0.2) == [2,2,3,4,5,6,7,7]
+@test winsor!([1,2,3,4,5,6,7,8,9], prop=0.4) == [4,4,4,4,5,6,6,6,6]
+@test winsor!([1,2,3,4,5,6,7,8],   count=1)  == [2,2,3,4,5,6,7,7]
+@test winsor!([1,2,3,4,5,6,7,8,9], count=3)  == [4,4,4,4,5,6,6,6,6]
+
+@test_throws ArgumentError winsor!([])
+@test_throws ArgumentError winsor!([1,2,3,4,5], prop=0.5)
+
+### Variance
```

Comment:
We should also have at least one test of `mean(trim(...))` just to check that the combination of both functions works.